### ISSUE TITLE
CA1303 throws error when text has colon

### ISSUE THREAD

**Issue by paulomorgado** [dotnet/roslyn-analyzers#6288]
### Analyzer

**Diagnostic ID**: [CA1303](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1303): `Do not pass literals as localized parameters`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Versions**: 6.0.0, 7.0.0-preview1.22518.1

**Visual Studio Version**: 17.4.1+33110.190

### Describe the bug

If the text has a colon, an error with the text after the colon.

### Steps To Reproduce

1. Create a Visual Basic Class Library (.NET Framework) project.
2. Change `Class1` to:
```vb
Public Class Class1

    Public Sub New()
        Me.Text = "a error: BANG!"
    End Sub

    <ComponentModel.Localizable(True)>
    Property Text As String

End Class
```
4. Add the Microsoft.CodeAnalysis.NetAnalyzers NuGet package.
5. Set the severity of CA1303 to Warning
6. Rebuild

### Expected behavior

CA1303 warning.

### Actual behavior

Visual Studio error:

| Severity | Code | Description | Project     | File | Line                                | Suppression State |
|----------|------|-------------|-------------|------|-------------------------------------|-------------------|
| Error    |      | BANG!"      | ClassLibrary1 |      | C:\Temp\ClassLibrary1\ClassLibrary1\VBC |                   |

* There's no error in the output window.
* There's no C:\Temp\WindowsApp1\WindowsApp1\VBC file.
* Sometimes, consecutive builds do not produce the error.

**Comment by paulomorgado** [comment]
Test to reproduce:

```csharp
        [Fact]
        public async Task ParameterWithLocalizableAttribute_StringLiteralWithColonArgument_Method_DiagnosticAsync()
        {
            await VerifyCS.VerifyAnalyzerAsync(@"
using System.ComponentModel;

public class C
{
    public void M([LocalizableAttribute(true)] string param)
    {
    }
}

public class Test
{
    public void M1(C c)
    {
        var str = ""a:colon"";
        c.M(str);
    }
}
",
            // Test0.cs(16,13): warning CA1303: Method 'void Test.M1(C c)' passes a literal string as parameter 'param' of a call to 'void C.M(string param)'. Retrieve the following string(s) from a resource table instead: "a:colon".
            GetCSharpResultAt(16, 13, "void Test.M1(C c)", "param", "void C.M(string param)", "a:colon"));

            await VerifyVB.VerifyAnalyzerAsync(@"
Imports System.ComponentModel

Public Class C
    Public Sub M(<LocalizableAttribute(True)> param As String)
    End Sub
End Class

Public Class Test
    Public Sub M1(c As C)
        Dim str = ""a:colon""
        c.M(str)
    End Sub
End Class
",
            // Test0.vb(12,13): warning CA1303: Method 'Sub Test.M1(c As C)' passes a literal string as parameter 'param' of a call to 'Sub C.M(param As String)'. Retrieve the following string(s) from a resource table instead: "a:colon".
            GetBasicResultAt(12, 13, "Sub Test.M1(c As C)", "param", "Sub C.M(param As String)", "a: colon"));
        }
```

The test passes, though.

**Comment by paulomorgado** [comment]
Fails when building in the command-line:

```
$ dotnet msbuild -t:rebuild .\ClassLibrary1\
MSBuild version 17.4.0+18d5aef85 for .NET
C:\Temp\"a error : BANG!"\ClassLibrary1\Class1.vb(4,9): warning CA1303: Method 'Sub Class1.New()' passes a literal string as parameter 'Au
toPropertyValue' of a call to 'Property Set Class1.Text(AutoPropertyValue As String)'. Retrieve the following string(s) from a resource table instead: "a error
 : BANG!". [C:\Temp\ClassLibrary1\ClassLibrary1\ClassLibrary1.vbproj]
VBC : Me.Text = "a error : BANG!" [C:\Temp\ClassLibrary1\ClassLibrary1\ClassLibrary1.vbproj]
```

**Comment by mavasani** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/37508
